core/arch/arm/plat-stm32mp1/main.c

```diff
@@ -329,3 +334,40 @@ unsigned int stm32_get_gpio_bank_clock(unsigned int bank)
 	assert(bank <= GPIO_BANK_K);
 	return GPIOA + bank;
 }
+
+#ifdef CFG_STM32_RNG
+#define RNG1_RESET_TIMEOUT_US		1000
+#define PRNG_SEED_SIZE			16
+
+void plat_rng_init(void)
+{
+	vaddr_t rng = (vaddr_t)phys_to_virt(RNG1_BASE, MEM_AREA_IO_SEC);
+	vaddr_t rcc = stm32_rcc_base();
+	uint64_t timeout_ref = timeout_init_us(RNG1_RESET_TIMEOUT_US);
+	uint8_t seed[PRNG_SEED_SIZE] = { };
+	size_t size = 0;
+
+	assert(cpu_mmu_enabled());
```

Comment:
How about asserting that the `phys_to_virt()` calls succeded for `rng` and `rcc` instead?